src/api/indexedDb.ts

```diff
@@ -103,45 +161,37 @@ export class IndexedDB {
 	}
 
 	private async initDb() {
+		const dbVersion = localStorageWorker.getDBVersion();
 		this.db = await openDB<TH2DB>(this.env, dbVersion, {
-			async upgrade(db, oldVersion) {
-				if (oldVersion === 0) {
-					const eventsStore = db.createObjectStore(IndexedDbStores.EVENTS, { keyPath: 'id' });
-					eventsStore.createIndex('timestamp', 'timestamp');
-
-					const messagesStore = db.createObjectStore(IndexedDbStores.MESSAGES, {
-						keyPath: 'id',
-					});
-					messagesStore.createIndex('timestamp', 'timestamp');
-
-					const searchHistoryStore = db.createObjectStore(IndexedDbStores.SEARCH_HISTORY, {
-						keyPath: 'timestamp',
-					});
-					searchHistoryStore.createIndex('timestamp', 'timestamp');
-
-					const graphSearchHistoryStore = db.createObjectStore(
-						IndexedDbStores.GRAPH_SEARCH_HISTORY,
-						{
-							keyPath: 'id',
-						},
-					);
-					graphSearchHistoryStore.createIndex('timestamp', 'timestamp');
-
-					const messageDisplayRulesStore = db.createObjectStore(IndexedDbStores.DISPLAY_RULES, {
-						keyPath: 'id',
-					});
-					messageDisplayRulesStore.createIndex('timestamp', 'timestamp');
-
-					const messageBodySortOrderStore = db.createObjectStore(
-						IndexedDbStores.MESSAGE_BODY_SORT_ORDER,
-						{
-							keyPath: 'id',
-						},
-					);
-					messageBodySortOrderStore.createIndex('timestamp', 'timestamp');
-				}
+			async upgrade(db) {
+				Object.entries(indexedDBkeyPaths).forEach(([storeName, keyPath]) => {
```

Comment:
This will recreate all the stores every dbversion update, which is not what we want